docs/Design.md

```diff
@@ -352,21 +352,35 @@ so it makes sense to have a disambiguator.
 
 ### Symbol names for declarations
 
+<!-- NOTE(def: symbol-names-for-decls) -->
+
 - For builtins, we can create a synthetic namespace e.g. `$builtin::type`.
   (using `$` because it is not legal in C++)
 - For FQNs with no anonymous namespaces, we will use the FQN directly.
 - For anonymous namespaces in FQNs, they should be replaced with the TU path,
   and some suffix like `$anon`.
   Anonymous namespaces in headers are not semantically meaningful,
   we need to use the path of the final TU, not the path of the header.
+- For anonymous types defined in headers, we need to use the header path
+  in the symbol name, so that we can consistently identify the same
+  anonymous type even when the header is included in different TUs.
```

Comment:
Question, not feedback: C++, so many edge cases. Do people do things like #include the same file in different namespaces, in the same TU or different TUs? In general how do you have a sense of "this is whacky but important" versus "this is whacky and we can ignore it?"

Reply:
> Do people do things like #include the same file in different namespaces, in the same TU or different TUs?

Yes. There's some kinds of files which are meant for inclusion like this. For example, in LLVM, there are tablegen files: https://sourcegraph.com/github.com/llvm/llvm-project/-/blob/clang/include/clang/Basic/DeclNodes.td -- these get processed into `.inc` files which are included inside nested contexts (E.g. https://sourcegraph.com/github.com/llvm/llvm-project/-/blob/clang/lib/AST/Decl.cpp?L1791:21)

> In general how do you have a sense of "this is whacky but important" versus "this is whacky and we can ignore it?"

My general rule of thumb for weird stuff is that we should support it as much as possible, because given the amount of C++ code that people have, chances are some weird functionality is being used at least somewhere important. If we can't support something, we should document it, at least in a comment in the code. Ideally, we'd have user-facing documentation (e.g. a `KnownLimitations.md`), but things are still moving around a bit, so I've not added that.

For specific queries like this, we can sometimes get (approximate) answers using Sourcegraph. E.g. https://sourcegraph.com/search?q=context:global+lang:C%2B%2B+namespace+:%5B%7E%28a-zA-Z_%29*%5D%7B...%23include...%7D&patternType=structural&sm=1&groupBy=repo